pkg/proxy/switch.go

```diff
@@ -425,6 +417,17 @@ func (s *SwitchSession) Bridge(userConn UserConnection, srvConn srvconn.ServerCo
 		lastActiveTime = time.Now()
 	}
 }
+func (s *SwitchSession) handleDisconnection(room *exchange.Room, parser *Parser, replayRecorder *ReplyRecorder, msg string) {
+	msg = utils.WrapperWarn(msg)
+	replayRecorder.Record([]byte(msg))
+
+	roomMessage := &exchange.RoomMessage{Event: exchange.DataEvent, Body: []byte("\n\r" + msg)}
+	if parser.zmodemParser.Status() != "" {
+		roomMessage.Body = append(zmodem.SkipSequence, zmodem.CancelSequence...)
+	}
+
+	room.Broadcast(roomMessage)
+}
 
 func (s *SwitchSession) recordSessionFinished(reason model.SessionLifecycleReasonErr) {
 	logObj := model.SessionLifecycleLog{Reason: string(reason)}
```

Comment:
I can only provide an English response. Please let me know if you need any specific feedback on the provided JavaScript code snippet.